src/modules/kazoo/kazoo.c

```diff
@@ -435,27 +487,46 @@ static int fire_init_event(int rank)
 	struct sip_msg *fmsg;
 	struct run_act_ctx ctx;
 	int rtb, rt;
+	sr_kemi_eng_t *keng = NULL;
 
 	LM_DBG("rank is (%d)\n", rank);
 	if (rank!=PROC_INIT)
 		return 0;
 
-	rt = route_get(&event_rt, "kazoo:mod-init");
-	if(rt>=0 && event_rt.rlist[rt]!=NULL) {
-		LM_DBG("executing event_route[kazoo:mod-init] (%d)\n", rt);
-		if(faked_msg_init()<0)
-			return -1;
-		fmsg = faked_msg_next();
-		rtb = get_route_type();
-		set_route_type(REQUEST_ROUTE);
-		init_run_actions_ctx(&ctx);
-		run_top_route(event_rt.rlist[rt], fmsg, &ctx);
-		if(ctx.run_flags&DROP_R_F)
-		{
-			LM_ERR("exit due to 'drop' in event route\n");
-			return -1;
+	if (kazoo_route_no>=0) {
+		rt = route_get(&event_rt, "kazoo:mod-init");
+        if(rt>=0 && event_rt.rlist[rt]!=NULL) {
+			LM_DBG("executing event_route[kazoo:mod-init] (%d)\n", rt);
+			if(faked_msg_init()<0)
+				return -1;
+			fmsg = faked_msg_next();
+			rtb = get_route_type();
+			set_route_type(REQUEST_ROUTE);
+			init_run_actions_ctx(&ctx);
+			run_top_route(event_rt.rlist[rt], fmsg, &ctx);
+			if(ctx.run_flags&DROP_R_F)
+			{
+				LM_ERR("exit due to 'drop' in event route\n");
+				return -1;
+			}
+			set_route_type(rtb);
+		}
+	} 
+  else {
+		keng = sr_kemi_eng_get();
+		if(keng!=NULL) {
+			sip_msg_t *msg;
+			str evrtname = str_init("kazoo:mod-init");
+			rtb = get_route_type();
+			msg = faked_msg_next();
+			if(sr_kemi_route(keng, msg, EVENT_ROUTE, &kazoo_event_callback, &evrtname)<0) {
+				LM_ERR("error running event route kemi callback\n");
+		  }
+		  set_route_type(rtb);
+		} 
+    else {
+			LM_ERR("no event route or kemi callback found for execution\n");
```

Comment:
split this in two different function that are evoked by `fire_init_event` depending on `kemi` being active

Reply:
Now depending on using cfg or kemi:
     - fire_init_event() calls fire_init_event_cfg() or fire_init_event_kemi()